src/client/opamPinCommand.ml

```diff
@@ -482,9 +485,13 @@ and source_pin
         "Couldn't retrieve opam file from versioned source, \
          using the one found locally.";
       Some local
-    | Some local, Some vers when not (OpamFile.OPAM.effectively_equal local vers) ->
+    | Some local, Some vers when
+        not OpamFile.(OPAM.effectively_equal
+                        (OPAM.with_url URL.empty local)
+                        (OPAM.with_url URL.empty vers)) ->
       OpamConsole.warning
-        "Opam file have significant uncommitted changes, using the versioned one";
+        "%s opam file have significant uncommitted changes, using the versioned one"
```

Comment:
`%s's opam file has non-trivial…` ?
"significant" could be interpreted as "large" in this context, which is misleading.